docs/en/enterprise-edition/policy-reference/aws-policies/s3-policies/bc-aws-s3-24.adoc

```diff
@@ -17,63 +17,52 @@
 
 |Subtype
 |Build
-//, Run
 
 |Frameworks
 |Terraform,TerraformPlan
 
 |=== 
 
 
-
 === Description 
 
 
-If a bucket's versioning configuration is MFA Delete: enabled, the bucket owner must include the x-amz-mfa request header to delete an object.
-Requests that include x-amz-mfa must use HTTPS.
-Configuring a bucket to enable MFA (multi-factor authentication) Delete requires additional authentication for either of the following operations:  (1) Change the versioning state of your bucket  (2) Permanently delete an object version.
-We recommend you configure S3 bucket with MFA Delete: enabled.
-
-////
-=== Fix - Runtime
-
-
-* CLI Command* 
-
-
-
-
-[source,shell]
-----
-{
- "aws s3api put-bucket-versioning --profile my-root-profile --bucket my-bucket-name --versioning-configuration Status=Enabled,MFADelete=Enabled --mfa "arn:aws:iam::00000000:mfa/root-account-mfa-device 123456"",
+This policy is designed to prevent configuration policies that could potentially lock out non-root users from accessing S3 buckets. This check is vital for maintaining access control in AWS environments and ensuring that bucket policies do not inadvertently create security risks by overly restrictive configurations.
 
-}
-----
-////
 
 === Fix - Buildtime
 
 
 *Terraform* 
 
 
-* *Resource:* aws_s3_bucket
-* *Argument:* mfa_delete - (Optional) Enable MFA delete for either Change the versioning state of your bucket or Permanently delete an object version.
-
-Default is false.
-This cannot be used to toggle this setting but is available to allow managed buckets to reflect the state in AWS
+* *Resource:* aws_s3_bucket, aws_s3_bucket_policy
+* *Argument:* policy
 
+Do not allow all actions, principals, and resources in an S3 policy.
```

Comment:
If this means: S3 policies should not grant permissions for all possible actions, to all users, on all resources within an S3 bucket, I recommend: "Avoid granting unrestricted access to all actions, users, and resources in an S3 policy"

Reply:
It's the inverse. It's make sure you don't set the effect to DENY and the action overly broad and the resource set to all.